src/mimeparser/mimeparser_tests.rs

```diff
@@ -2011,3 +1987,51 @@ async fn test_protected_date() -> Result<()> {
     assert_eq!(alice_msg.get_timestamp(), bob_msg.get_timestamp());
     Ok(())
 }
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+async fn test_huge_image_becomes_file() {
+    let t = TestContext::new_alice().await;
+    receive_imf(
+        &t.ctx,
+        include_bytes!("../../test-data/message/image_huge_36M.eml"),
+        false,
+    )
+    .await
+    .unwrap();
+    let chats = Chatlist::try_load(&t.ctx, 0, None, None).await.unwrap();
+    let msg_id = chats.get_msg_id(0).unwrap().unwrap();
+    let msg = Message::load_from_db(&t.ctx, msg_id).await.unwrap();
+    assert_eq!(msg.text, "Large image test – Here is a huge image");
```

Comment:
I'd remove all checks that are not related to the fix, particularly for text, is_error(), is_dc_message, chat_blocked, state. They just add noise, we don't need to test all `Message` fields in all tests. And e.g. if we decide later to change En dash after the subject to Em dash (btw, i wonder why En dash is used), we need to fix many tests which is inconvenient. It'd be better to check `get_width()` and `get_height()` which must behave differently in the two added tests.